src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexChannelFactoryTest.cs

```diff
@@ -101,7 +101,8 @@ public static void CreateChannel_BasicHttpBinding_Throws_InvalidOperation()
             factory.CreateChannel();
         });
 
-        Assert.True(exception.Message.Contains("BasicHttpBinding"), "InvalidOperationException exception string should contain 'BasicHttpBinding'");
+        Assert.True(exception.Message.Contains("BasicHttpBinding"), 
+            string.Format("InvalidOperationException exception string should contain 'BasicHttpBinding'. Actual message:\r\n" + exception.ToString()));
     }
```

Comment:
Fair enough - I documented the anomaly in line 175 now. In general we always want to check for the exception message if possible.